src/pytest_memray/marks.py

```diff
@@ -225,14 +237,22 @@ def limit_leaks(
     location_limit: str,
     *,
     filter_fn: Optional[LeaksFilterFunction] = None,
+    current_thread_only: bool = False,
     _result_file: Path,
     _config: Config,
     _test_id: str,
 ) -> _LeakedInfo | None:
     reader = FileReader(_result_file)
-    allocations: list[AllocationRecord] = list(
-        reader.get_leaked_allocation_records(merge_threads=True)
-    )
+    allocations: list[AllocationRecord] = []
+    if current_thread_only:
+        main_thread_id = reader.metadata.main_thread_id
+        allocations.extend(
+            record
+            for record in reader.get_leaked_allocation_records(merge_threads=False)
+            if record.tid == main_thread_id
+        )
+    else:
+        allocations.extend(reader.get_leaked_allocation_records(merge_threads=True))
```

Comment:
Likewise, could be simplified to
```suggestion
    allocations: list[AllocationRecord] = [
        record
        for record in reader.get_leaked_allocation_records(
            merge_threads=not current_thread_only
        )
        if not current_thread_only or record.tid == reader.metadata.main_thread_id
    ]
```